### PR TITLE
Add JSI Advanced Transparent Pods from SpaceDock

### PR DESCRIPTION
**Pull request by Space-Duck**
This pull request was automatically generated by SpaceDock on behalf of Jplrepo, to add [JSI Advanced Transparent Pods](https://spacedock.info/mod/598/JSI%20Advanced%20Transparent%20Pods) to CKAN.

Mod details:
    name = /mod/598/JSI%20Advanced%20Transparent%20Pods
    author = Jplrepo
    abstract = This mod, previously called JSITransparentPods, as distributed as an integral part of the RasterPropMonitor (RPM) mod, has been split off from RPM and updated to an advanced version with futher development to come.
    license = GNU GPLv3 - https://github.com/JPLRepo/JSIAdvTransparentPods/blob/master/LICENSE
    Homepage = 
    description = [[http://imgur.com/a/Q7ZNd#0]]

TThis mod contains a Part Module that allows Part Transparency (ability to see inside the parts to the IVAs) in the editor and in flight.   
It used to be called JSITransparentPod and was distributed as part of RasterPropMonitor bundle. MOARdV and I have agreed to split it off from the main RPM and from now on will be distributed separately.   
Yes Stock KSP 1.1 has a kind of IVA Overlay feature. But it does not go all the way that this module does.   
   
This module is an update to the previous module and will be known as JSIAdvTransparentPods.   
**On it's own this mod will do nothing unless you have also installed mods that already support it or create your own part config files, or Module Manager files.
You can create MM files to make pretty much any and all parts that have internals be transparent, but they are not included in this download. **  
Features of this module include:   
* Transparency feature for configured parts that will show the IVA internal models of parts that have been configured to use this module.   
* Ability to Swap Shaders on the External and Internal model to make Window appear Opaque or Transparent.   
* Ability to toggle Transparency setting ON/OFF/AUTO (mouse over) on a part by part basis.   
* Transparency (show internals) for ALL Loaded vessels (not just the Active Vessel).   
* Automatically handles and supports when the user Toggles the stock IVA Overlay feature on and off.

**Future Enhancements:**   
Ability to swap Depth Shader Meshes, to support window transparency available during normal JSIAdvTransparentPod operations and a larger cut-out view when the stock Overlay is turned on.   
   
**Known Issues:**
Due to Unity camera limitations and the way KSP camera and universe is set-up (It's complicated, but if you want the full explanation send me a PM) any objects in the foreground of an IVA can be occluded/cut-off by the IVA.
This also occurs with the stock overlay feature introduced in KSP 1.1.
I have implemented some basic ray casting to help allieviate this, but it only works when the object in front of an IVA is directly between the IVA and the player's viewing camera.
I am investigating other ways to deal with this issue, such as shaders and depth-buffer, hand-wave algorithms.
   
For mod authors you need to follow the specific requirements of this mod in your models and config files.    
For details on that refer to the wiki site page:   
https://github.com/JPLRepo/JSIAdvTransparentPods/wiki/JSIAdvTransparentPods   

**Support:**
NO LOG = NO SUPPORT.
Whilst I agree CKAN is a great mod for those that can't use Zip tools. I take no part, nor am I interested in maintaining the CKAN mod metadata for my mods.
Please don't ask me about it but refer to the CKAN mod thread if you are having issues with CKAN or the metadata it maintains.

Please direct questions about this pull request to [Jplrepo](https://spacedock.info/profile/Jplrepo).

### CODE REVIEW

**Comment by plague006** [comment]
tl;dr Waiting on new RPM release.

So this is/was a part of RPM

>It used to be called JSITransparentPod and was distributed as part of RasterPropMonitor bundle. MOARdV and I have agreed to split it off from the main RPM and from now on will be distributed separately.

So presumably we'll have mutual `recommends` this mod. However at the moment RPM has not been updated and they both include `JSIAdvTransparentPods` so they currently conflict. Presumably a new RPM release will be coming soon and until that happens we're stuck on this.

**Comment by politas** [comment]
#rebuild

**Comment by plague006** [comment]
Closed by https://github.com/KSP-CKAN/NetKAN/pull/3992
